internal/rule/name.go

```diff
@@ -0,0 +1,137 @@
+// Package rule defines utility functions some revive rules can share.
+package rule
+
+import (
+	"strings"
+	"unicode"
+)
+
+// commonInitialisms is a set of common initialisms.
+// Only add entries that are highly unlikely to be non-initialisms.
+// For instance, "ID" is fine (Freudian code is rare), but "AND" is not.
+var commonInitialisms = map[string]bool{
+	"ACL":   true,
+	"API":   true,
+	"ASCII": true,
+	"CPU":   true,
+	"CSS":   true,
+	"DNS":   true,
+	"EOF":   true,
+	"GUID":  true,
+	"HTML":  true,
+	"HTTP":  true,
+	"HTTPS": true,
+	"ID":    true,
+	"IDS":   true,
+	"IP":    true,
+	"JSON":  true,
+	"LHS":   true,
+	"QPS":   true,
+	"RAM":   true,
+	"RHS":   true,
+	"RPC":   true,
+	"SLA":   true,
+	"SMTP":  true,
+	"SQL":   true,
+	"SSH":   true,
+	"TCP":   true,
+	"TLS":   true,
+	"TTL":   true,
+	"UDP":   true,
+	"UI":    true,
+	"UID":   true,
+	"UUID":  true,
+	"URI":   true,
+	"URL":   true,
+	"UTF8":  true,
+	"VM":    true,
+	"XML":   true,
+	"XMPP":  true,
+	"XSRF":  true,
+	"XSS":   true,
+}
+
+// Name returns a different name of struct, var, const, or function if it should be different.
+// Currently, the function is used only for the `var-naming` rule.
```

Comment:
This is the kind of comment I would discourage.

`Currently` is meaningful when it's written, but instantly becomes outdated. A date would be better.

Also, the sentence is lost in the middle, not even it's own paragraph.

It's an exported go doc, so it will present in the package documentation. It shouldn't be visible there.

Then if there is a future refactoring, I'm pretty sure either this sentence would be forgotten if someone reuses Name method.

I would use it in the commit message and remove it here.

Yes, it's a long post about a single line of comment 😁